common/src/main/java/com/wynntils/functions/MinecraftFunctions.java

```diff
@@ -59,4 +64,46 @@ public FunctionArguments.Builder getArgumentsBuilder() {
                     List.of(new FunctionArguments.Argument<>("keyCode", Integer.class, null)));
         }
     }
+
+    public static class PlayerEffectFunction extends Function<Integer> {
+        @Override
+        public Integer getValue(FunctionArguments arguments) {
+            String effectName = arguments.getArgument("effectName").getStringValue();
+            ResourceLocation effectLocation = ResourceLocation.withDefaultNamespace(effectName);
+            MobEffect effect = BuiltInRegistries.MOB_EFFECT.get(effectLocation);
+
+            if (effect == null) {
+                return -1; // Effect not found
+            }
+
+            Holder<MobEffect> effectHolder =
+                    BuiltInRegistries.MOB_EFFECT.getHolder(effectLocation).orElse(null);
+
+            if (effectHolder == null) {
+                return -1; // Effect holder not found
+            }
+
+            // Check if the player has the effect
+            if (McUtils.player().hasEffect(effectHolder)) {
+                MobEffectInstance effectInstance = McUtils.player().getEffect(effectHolder);
+
+                if (effectInstance != null && effectInstance.getDuration() > 0) {
```

Comment:
Maybe >= 0 so in that 1 second where it displays 0, we can still get a valid number